test/logic_value_test.dart

```diff
@@ -1022,4 +1025,97 @@ void main() {
           equals(LogicValue.filled(0, LogicValue.zero).hashCode));
     });
   });
+
+  group('random value generation ', () {
+    test('should throw exception when max is not int or BigInt.', () {
+      expect(() => Random(5).nextLogicValue(width: 10, max: '10'),
+          throwsA(isA<InvalidRandomLogicValueException>()));
+      expect(() => Random(5).nextLogicValue(width: 10, max: 10.5),
+          throwsA(isA<InvalidRandomLogicValueException>()));
+    });
+    test('should throw exception when max is less than 0.', () {
+      expect(() => Random(5).nextLogicValue(width: 10, max: -1),
+          throwsA(isA<InvalidRandomLogicValueException>()));
+      expect(() => Random(5).nextLogicValue(width: 10, max: BigInt.from(-10)),
+          throwsA(isA<InvalidRandomLogicValueException>()));
+    });
+    test(
+        'should throw exception when max is set when generate random num with'
+        ' invalid bits.', () {
+      expect(
+          () => Random(5).nextLogicValue(
+              width: 10,
+              includeInvalidBits: true,
+              max: LogicValue.ofInt(10, 10)),
+          throwsA(isA<InvalidRandomLogicValueException>()));
+    });
+    test(
+        'should return random logic value with invalid bits when '
+        'includeInvalidBits is set to true.', () {
+      final lvRand =
+          Random(5).nextLogicValue(width: 10, includeInvalidBits: true);
+
+      expect(lvRand.toString(), contains('x'));
+    });
+    test('should return empty LogicValue when width is 0.', () {
+      expect(Random(5).nextLogicValue(width: 0), equals(LogicValue.empty));
+    });
+    test(
+        'should return valid results when using BigInt as max '
+        'and integer width.', () {
+      for (var i = 0; i <= 64; i++) {
+        final randMaxBigInt = Random(5)
+            .nextLogicValue(width: i, max: BigInt.parse('9999999999999999999'));
+        final randMaxInt =
+            Random(5).nextLogicValue(width: i, max: BigInt.from(30));
+        expect(randMaxBigInt, equals(Random(5).nextLogicValue(width: i)));
+        expect(randMaxInt.toInt(), lessThan(30));
+      }
+    });
+    test(
+        'should return valid results when using int as max '
+        'and BigInt width.', () {
```

Comment:
what about `int` max when width <= 64?

Reply:
Ya, its in here:

```dart
    test(
        'should return random int logic value without invalid bits when'
        ' having different width and max constraint.', () {
```

Merged to avoid confusion.